node/node.go

```diff
@@ -437,17 +455,25 @@ func (node *Node) ReliableCallObject(
 
 	if err != nil {
 		node.logger.Errorf("Failed to insert or get reliable call %s: %v", callID, err)
-		return nil, fmt.Errorf("failed to insert or get reliable call: %w", err)
+		return nil, goverse_pb.ReliableCallStatus_UNKNOWN, fmt.Errorf("failed to insert or get reliable call: %w", err)
 	}
 
 	// Check if the call was already completed (idempotency)
+	status := stringToReliableCallStatus(rc.Status)
 	switch rc.Status {
 	case "success":
 		node.logger.Infof("Reliable call %s already succeeded, returning cached result", callID)
-		return protohelper.BytesToAny(rc.ResultData)
-	case "failed":
-		node.logger.Infof("Reliable call %s already failed, returning cached error", callID)
-		return nil, fmt.Errorf("reliable call failed: %s", rc.Error)
+		resultAny, err := protohelper.BytesToAny(rc.ResultData)
+		if err != nil {
+			// Deserialization failed for a successful call - this shouldn't happen
+			// Return UNKNOWN status to indicate an internal error
+			node.logger.Errorf("Failed to deserialize result for successful call %s: %v", callID, err)
+			return nil, goverse_pb.ReliableCallStatus_UNKNOWN, fmt.Errorf("failed to deserialize successful call result: %w", err)
```

Comment:
in such case we should return FAILED. At least the user knows the function was executed.